### PR TITLE
adds `prototype` to stickybits

### PR DESCRIPTION
**Pull request by yowainwright**

### CODE REVIEW

**Comment by briangonzalez** [comment]
High level, what should I be looking for here?

**Comment by yowainwright** [comment]
@briangonzalez switches `stickybits` to prototype pattern—the main benefit being "more easily ensure that the window scroll event happens less"

File: src/stickybits.js [hunk 1]
```diff
@@ -40,27 +52,41 @@ function Stickybit(target, o) {
...
+  let invoked;
+  const fps = 1000 / 60;
+  function setupStickiness() {
+    stickiness();
+    invoked = false;
+  }
+  function checkStickiness() {
+    if (invoked) return;
+    invoked = true;
+    window.setTimeout(setupStickiness, fps);
```
Comment (jonathanong) [inline src/stickybits.js]: `setTimeout(() => {}, 0)`

File: src/stickybits.js [hunk 2]
```diff
@@ -40,27 +52,41 @@ function Stickybit(target, o) {
...
-      return;
     }
-    return;
   }
-  scrollTarget.addEventListener('scroll', () => scrollTarget.requestAnimationFrame(stickiness));
-}
+  let invoked;
+  const fps = 1000 / 60;
+  function setupStickiness() {
+    stickiness();
```
Comment (jonathanong) [inline src/stickybits.js]:
```js
if (invoked) return

stickiness();
invoked = true
```

**Pull request merged**